chatcommands.py

```diff
@@ -0,0 +1,935 @@
+from globalvars import GlobalVars
+from findspam import FindSpam
+from datetime import datetime
+from utcdate import UtcDate
+from apigetpost import api_get_post
+from datahandling import *
+from chatcommunicate import *
+from metasmoke import Metasmoke
+from parsing import *
+from spamhandling import handle_spam
+from spamhandling import handle_user_with_all_spam
+from threading import Thread
+import random
+import requests
+import os
+import time
+
+
+# TODO: pull out code block to get user_id, chat_site, room_id into function
+# TODO: Return result for all functions should be similar (tuple/named tuple?)
+# TODO: Do we need uid == -2 check?  Turn into "is_user_valid" check
+# TODO: Consistant return structure
+#   if return...else return vs if return...return
+
+# Functions go before the final dictionaries of command to function mappings
+
+def is_report(post_site_id):
+    """
+    Checks if a post is a report
+    :param post_site_id: Report to check
+    :return: Boolean stating if it is a report
+    """
+    if post_site_id is None:
+        return False
+    return True
+
+
+def send_metasmoke_feedback(post_url, second_part_lower, ev_user_name, ev_user_id):
+    """
+    Sends feedback to MetaSmoke
+    :param post_url: The post url we are sending
+    :param second_part_lower: Feedback
+    :param ev_username: User name supplying the feedback
+    :param ev_user_id: User ID supplying the feedback
+    :return: None
+    """
+    t_metasmoke = Thread(target=Metasmoke.send_feedback_for_post,
+                         args=(post_url, second_part_lower, ev_user_name, ev_user_id,))
+    t_metasmoke.start()
+
+
+def single_random_user(ev_room):
+    """
+    Returns a single user name from users in a room
+    :param ev_room: Room to select users from
+    :return: A single user tuple
+    """
+    return random.choice(GlobalVars.users_chatting[ev_room])
+
+#
+#
+# System command functions below here
+# Each of these should take the *args and **kwargs parameters. This allows us to create functions that
+# don't accept any parameters but still use the `command_dict` mappings
+
+
+# --- Blacklist Functions --- #
+def command_add_blacklist_user(*args, **kwargs):
+    """
+    Adds a user to the site blacklist
+    :param kwargs: Requires that 'content_lower', 'message_url', 'ev_room', 'ev_user_id', and 'wrap2' is passed as kwarg
+    :return: A string
+    """
+    if is_privileged(kwargs['ev_room'], kwargs['ev_user_id'], kwargs['wrap2']):
```

Comment:
Not each function takes the exact same arguments. By passing the dictionary, instead of explicitly setting each function's parameters, I can call it like I do here:

https://github.com/AWegnerGitHub/SmokeDetector/blob/branch_chatcommands_split/chatcommunicate.py#L222

```
    # Process additional commands
    command_parameters = {
        'content': content,
        'content_lower': content_lower,
        'ev_room': ev_room,
        'ev_room_name': ev_room_name,
        'ev_user_id': ev_user_id,
        'ev_user_name': ev_user_name,
        'message_parts': message_parts,
        'message_url': message_url,
        'wrap2': wrap2,
    }
    try:
        return cmds[command](**command_parameters)
    except KeyError:
        return False, None  # Unrecognized command, can be edited later.
```

The dictionary-switch statement (because Python lacks a true `switch`) allows us to [map the commands to desired function](https://github.com/AWegnerGitHub/SmokeDetector/blob/branch_chatcommands_split/chatcommands.py#L848). This allows us to quickly add aliases for commands (ie. `commands`, `help`, `info`)